internal/pkg/addon/package.go

```diff
@@ -0,0 +1,302 @@
+// TODO(dnrnd) AWS::CloudFormation::Stack
+var transformInfoFor = map[string][]transformInfo{
+	"AWS::ApiGateway::RestApi": {
+		{
+			Property:           []string{"BodyS3Location"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+		},
+	},
+	"AWS::Lambda::Function": {
+		{
+			Property:           []string{"Code"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+			ForceZip:           true,
+		},
+	},
+	"AWS::Lambda::LayerVersion": {
+		{
+			Property:           []string{"Content"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+			ForceZip:           true,
+		},
+	},
+	"AWS::AppSync::GraphQLSchema": {
+		{
+			Property: []string{"DefinitionS3Location"},
+		},
+	},
+	"AWS::AppSync::Resolver": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::AppSync::FunctionConfiguration": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::ElasticBeanstalk::ApplicationVersion": {
+		{
+			Property:           []string{"SourceBundle"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+		},
+	},
+	"AWS::Glue::Job": {
+		{
+			Property: []string{"Command", "ScriptLocation"},
+		},
+	},
+	"AWS::StepFunctions::StateMachine": {
+		{
+			Property:           []string{"DefinitionS3Location"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+		},
+	},
+	"AWS::CodeCommit::Repository": {
+		{
+			Property:           []string{"Code", "S3"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+			ForceZip:           true,
+		},
+	},
+}
+
+func (a *Addons) packageLocalArtifacts(tmpl *cfnTemplate) (*cfnTemplate, error) {
+	resources := mappingNode(&tmpl.Resources)
+
+	for name := range resources {
+		res := mappingNode(resources[name])
+		typeNode, ok := res["Type"]
+		if !ok || typeNode.Kind != yaml.ScalarNode {
+			continue
+		}
+
+		propsNode, ok := res["Properties"]
+		if !ok || propsNode.Kind != yaml.MappingNode {
+			continue
+		}
+
+		transforms, ok := transformInfoFor[typeNode.Value]
+		if !ok {
+			continue
+		}
+
+		for _, tr := range transforms {
+			if err := a.transformProperty(propsNode, tr); err != nil {
+				return nil, fmt.Errorf("transform property %s property for %s: %w", name, tr.Property, err)
+			}
+		}
+	}
+
+	return tmpl, nil
+}
+
+func (a *Addons) transformProperty(properties *yaml.Node, tr transformInfo) error {
+	mapNode := mappingNode(properties)
+	var node *yaml.Node
+	for i, key := range tr.Property {
+		var ok bool
+		node, ok = mapNode[key]
+		if !ok || (i+1 != len(tr.Property) && node.Kind != yaml.MappingNode) {
+			return nil // no error if the property doesn't exist
+		}
+		mapNode = mappingNode(node)
+	}
+
+	if node == nil || node.Kind != yaml.ScalarNode {
+		// only transform if the node is a scalar node
```

Comment:
Does this only happen when CFN changes their schema? should we give a warning instead? At least users would be aware of the malfunctioning and let us know